### PR TITLE
1.2.0 update (take 2)

### PR DESCRIPTION
**Pull request by danomatika**
This is a series of modernization updates and a couple features required by our usage at ZKM:

* bumped to version 1.2
* update to the pd-lib-builder makefile
* added vbap loudspeaker set info outlet which reports speaker triplets and current triplet
* lowered loudspeaker intersection epsilon for float-based math (should this be settable?)
* limit min spread to avoid NaNs
* updated help files and removed pddp links for now
* commented out a few extraneous logposts and various formatting fixes
* updated changelog

This is a new version of #1 without the submodule or subtree for pd-lib-builder.

### CODE REVIEW

**Comment by danomatika** [comment]
I've gone through the files to fix syntax & formatting and notice a couple of things:

* there is plenty of redundant code, especially between vbap.c & rvbap.c
* the Max build support makes things a bit clunky and a number of things are commented out/hacked when Pd support was added

@umlaeute Do you think it would be safe to remove the Max support at this point and make this "PD only?" I wonder if this specific code (ie. this copy) base was *ever* built for Max.

I could also imagine consolidating the general vector math and algorithm code into an extra .c file.

UPDATE: code clean up is in the `update/cleanup` branch.

**Comment by umlaeute** [comment]
> Do you think it would be safe to remove the Max support at this point and make this "PD only?"

honestly i cannot possibly say.

however, i think that much of the `#ifdef` foo could be replaced by proper helper macros/functions.

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,5 +1,19 @@
-14 Aug. 2014 update from version 1.0.3.2 to version  1.1
+2018 TBD version 1.2.0
```
Comment (umlaeute) [inline CHANGELOG.txt:1]: TBD? shouldn't that be `vbap`?

File: README.txt [hunk 2]
```diff
@@ -1,36 +1,48 @@
+# vbap
 
-A Pd port of the VBAP object for Max/MSP
+A Pure Data port of the VBAP object for Max/MSP
 
-Written by Ville Pulkki
-Helsinki University of Technology
-Laboratory of acoustics and audio signal processing
+Originally written by Ville Pulkki  
+Helsinki University of Technology
```
Comment (umlaeute) [inline README.txt]: nitpick, but of course i would strip the trailing whitespace.

**Review by umlaeute — COMMENTED** [review]
just attended to this long-standing PR-review.

File: CHANGELOG.txt (hunk 1 above)
Comment (danomatika) [reply CHANGELOG.txt:1]: TBD usually means "To Be Determined" and is a placeholder as I didn't know when the version would be finished/accepted.

**Comment by danomatika** [comment]
In looking at this again, I think the PDDP stuff should not have been removed. I can put it back.

File: CHANGELOG.txt (hunk 1 above)
Comment (danomatika) [reply CHANGELOG.txt:1]: > In looking at this again, I think the PDDP stuff should not have been removed. I can put it back.

Ah never mind. What I removed was just pddp links to other objects, not the meta files.

File: README.txt (hunk 2 above)
Comment (danomatika) [reply README.txt]: Done.

**Comment by danomatika** [comment]
I went ahead and did the formatting cleanup since previous commits had already touched on indentation.

UPDATE: removing redundant and out of date `update/cleanup` branch

**Comment by danomatika** [comment]
That should cover most of my updates. One question: what's the relationship of rvbap to vbap? It might be good to clarify this in the readme. It basically looks like a modified version of vbap as a good amount of code is similar.

**Comment by danomatika** [comment]
@umlaeute What do you think? Should we merge? I've been using this object for a while now at work (among other places).

**Comment by umlaeute** [comment]
so what does keep from merging?

**Comment by danomatika** [comment]
I think I was waiting on your review. I can merge now as I've been using it in a production app for a while now.

**Comment by danomatika** [comment]
Ok, merged and tagged.

**Pull request merged**